gpu/segmentation/include/pcl/gpu/segmentation/impl/gpu_extract_clusters.hpp

```diff
@@ -148,69 +148,84 @@ pcl::gpu::extractEuclideanClusters (const typename pcl::PointCloud<PointT>::Ptr
+        if (processed[idx])
           continue;
         processed[idx] = true;
         PointXYZ p;
         pcl::copyPoint((*host_cloud_)[idx], p);
-        queries_host.push_back (p);
+        queries_host.push_back(p);
         found_points++;
         r.indices.push_back(idx);
       }
-      PCL_DEBUG(" data.size: %i, foundpoints: %i, previous: %i", data.size() ,
-              found_points, previous_found_points);
-      PCL_DEBUG(" new points: %i, next queries size: %i\n", found_points - previous_found_points,
-             queries_host.size());
-    }
-    while (previous_found_points < found_points);
+      PCL_DEBUG(" data.size: %i, foundpoints: %i, previous: %i",
+                data.size(),
+                found_points,
+                previous_found_points);
+      PCL_DEBUG(" new points: %i, next queries size: %i\n",
+                found_points - previous_found_points,
+                queries_host.size());
+    } while (previous_found_points < found_points);
     // If this queue is satisfactory, add to the clusters
-    if (found_points >= min_pts_per_cluster && found_points <= max_pts_per_cluster)
-    {
-      std::sort (r.indices.begin (), r.indices.end ());
+    if (found_points >= min_pts_per_cluster && found_points <= max_pts_per_cluster) {
+      std::sort(r.indices.begin(), r.indices.end());
       // @todo: check if the following is actually still needed
-      //r.indices.erase (std::unique (r.indices.begin (), r.indices.end ()), r.indices.end ());
+      // r.indices.erase (std::unique (r.indices.begin (), r.indices.end ()),
+      // r.indices.end ());
 
       r.header = host_cloud_->header;
-      clusters.push_back (r);   // We could avoid a copy by working directly in the vector
+      clusters.push_back(r); // We could avoid a copy by working directly in the vector
     }
   }
 }
 
-template <typename PointT> void
-pcl::gpu::EuclideanClusterExtraction<PointT>::extract (std::vector<pcl::PointIndices> &clusters)
+template <typename PointT>
+void
+pcl::gpu::EuclideanClusterExtraction<PointT>::extract(
+    std::vector<pcl::PointIndices>& clusters)
 {
-/*
-  // Initialize the GPU search tree
-  if (!tree_)
-  {
-    tree_.reset (new pcl::gpu::Octree());
-    ///@todo what do we do if input isn't a PointXYZ cloud?
-    tree_.setCloud(input_);
-  }
-*/
-  if (!tree_->isBuilt())
-  {
+  /*
+    // Initialize the GPU search tree
+    if (!tree_)
+    {
+      tree_.reset (new pcl::gpu::Octree());
+      ///@todo what do we do if input isn't a PointXYZ cloud?
+      tree_.setCloud(input_);
+    }
+  */
+  if (!tree_->isBuilt()) {
     tree_->build();
   }
-/*
-  if(tree_->cloud_.size() != host_cloud.size ())
-  {
-    PCL_ERROR("[pcl::gpu::EuclideanClusterExtraction] size of host cloud and device cloud don't match!\n");
-    return;
-  }
-*/
+  /*
+    if(tree_->cloud_.size() != host_cloud.size ())
+    {
+      PCL_ERROR("[pcl::gpu::EuclideanClusterExtraction] size of host cloud and device
```

Comment:
Is this still correct?

Reply:
Him, this is a good question! This check seems to be active in the corresponding [cpu method](https://github.com/PointCloudLibrary/pcl/blob/d242fcbdbb53efc7de48c9159343432a2194a27c/segmentation/include/pcl/segmentation/extract_clusters.h#L112), but I am not sure this check is needed. I think constructing the tree with a superset of the pointcloud used for clustering should work. Nevertheless, I would prefer to activate it again in the GPU module for the sake of symmetry. What do you think?

Reply:
@larshg Thoughts?

(separate PR)